node/src/node_configurator/mod.rs

```diff
@@ -219,7 +219,8 @@ pub fn real_user_from_multi_config_or_populate(
     }
 }
 
-pub fn real_user_data_directory_opt_and_chain_name(
+#[allow(non_snake_case)]
+pub fn real_user__data_directory_opt__chain_name(
```

Comment:
Why the extra underscores? It forces you to use the `#[allow(non_snake_case)]` directive. It doesn't really communicate extra information on it's own. Is following the naming standard while including a comment better?